fahrschein/src/main/java/org/zalando/fahrschein/StreamParameters.java

```diff
@@ -136,7 +143,27 @@ public StreamParameters withStreamTimeout(int streamTimeout) throws IllegalArgum
         if(batchFlushTimeout != null && batchFlushTimeout > streamTimeout){
             throw new IllegalArgumentException("stream_timeout is lower than batch_flush_timeout.");
         }
-        return new StreamParameters(batchLimit, streamLimit, batchFlushTimeout, streamTimeout, streamKeepAliveLimit, maxUncommittedEvents);
+        return new StreamParameters(batchLimit, streamLimit, batchFlushTimeout, streamTimeout, streamKeepAliveLimit, maxUncommittedEvents, commitTimeout);
+    }
+
+    /**
+     * Maximum amount of seconds that nakadi will be waiting for commit after sending a batch to a client.
+     * In case if commit is not coming within this timeout, nakadi will initialize stream termination, no
+     * new data will be sent. Partitions from this stream will be assigned to other streams.
+     * Setting commit_timeout to 0 is equal to setting it to maximum allowed value - 60 seconds.
+     *
+     * @param commitTimeout
+     *          Stream initialization will fail if commit_timeout is lower than 0 or higher than 60.
+     * @throws IllegalArgumentException
+     */
+    public StreamParameters withCommitTimeout(int commitTimeout) throws IllegalArgumentException {
+        if(commitTimeout < 0){
+            throw new IllegalArgumentException("commit_timeout can't be lower than 0.");
+        }
+        if(commitTimeout > 60){
```

Comment:
I don't think we should be validating the maximum here since the setting on nakadi side seems to be configurable